prow/config/inrepoconfig.go

```diff
@@ -94,23 +95,61 @@ func defaultProwYAMLGetter(
 		return nil, fmt.Errorf("failed to merge: %v", err)
 	}
 
+	prowYAML := &ProwYAML{}
+
+	log.Debugf("Attempting to get %q.", inRepoConfigFileName)
 	prowYAMLFilePath := path.Join(repo.Directory(), inRepoConfigFileName)
-	if _, err := os.Stat(prowYAMLFilePath); err != nil {
-		if os.IsNotExist(err) {
-			log.Debugf("File %q does not exist.", inRepoConfigFileName)
-			return &ProwYAML{}, nil
+	if _, err := os.Stat(prowYAMLFilePath); err == nil {
+		log.Debugf("No error checking for %q, full path %q.", inRepoConfigFileName, prowYAMLFilePath)
+		bytes, err := ioutil.ReadFile(prowYAMLFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read %q: %v", inRepoConfigFileName, err)
 		}
-		return nil, fmt.Errorf("failed to check if file %q exists: %v", inRepoConfigFileName, err)
-	}
+		if err := yaml.Unmarshal(bytes, prowYAML); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal %q: %v", inRepoConfigFileName, err)
+		}
+	} else {
+		if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to check if file %q exists: %v", inRepoConfigFileName, err)
+		}
+		mergeProwYAML := func(a, b *ProwYAML) *ProwYAML {
+			c := &ProwYAML{}
+			c.Presets = append(a.Presets, b.Presets...)
 
-	bytes, err := ioutil.ReadFile(prowYAMLFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read %q: %v", inRepoConfigFileName, err)
-	}
+			c.Presubmits = append(a.Presubmits, b.Presubmits...)
 
-	prowYAML := &ProwYAML{}
-	if err := yaml.Unmarshal(bytes, prowYAML); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal %q: %v", inRepoConfigFileName, err)
+			c.Postsubmits = append(a.Postsubmits, b.Postsubmits...)
+
+			return c
+		}
+
+		log.Debugf("Attempting to read config files under %q.", inRepoConfigDirName)
+		prowYAMLDirPath := path.Join(repo.Directory(), inRepoConfigDirName)
+		err := filepath.Walk(prowYAMLDirPath, func(p string, info os.FileInfo, err error) error {
```

Comment:
Do we really want to make these two mutually exclusive?

Reply:
Talked with @chaodaiG about this and agreed to do it this way, fine with me if it is better to support both at the same time.

Reply:
Not able to observe a use case of when someone wants to use both `.prow.yaml` and `.prow` at the same time, so would rather enforce `.prow` for future to make our lives easier. I'm planning to further enforce this by a test in https://github.com/kubernetes/test-infra/tree/master/prow/cmd/checkconfig for any repo that enables inrepoconfig.